security/fma-multi-threaded-64-bit-cannon.md

```diff
@@ -0,0 +1,127 @@
+# [Project Name]: Failure Modes and Recovery Path Analysis
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Failure Modes and Recovery Paths](#failure-modes-and-recovery-paths)
```

Comment:
Are there any hard forks or contract upgrades involved? If so, let's make sure to consider any FMA-specific mitigations for the failure modes listed in [`fma-generic-hardfork.md`](https://github.com/ethereum-optimism/design-docs/blob/main/security/fma-generic-hardfork.md) and [`fma-generic-contracts.md`](https://github.com/ethereum-optimism/design-docs/blob/main/security/fma-generic-contracts.md)

Reply:
There isn't a contract upgrade. The onchain operation for this change requires changing the `DisputeGameFactory` implementation of cannon and permissioned_cannon. I have a section at the bottom that covers this operation. The other failure modes in the linked documents don't apply here.